### PR TITLE
Debug update june22

### PR DESCRIPTION
**Pull request by fredshone**
Overhauls the debug module, which was out of data and a massive pain. I am sharing with people who i think touch matsim configs such that they might make use of this.

This does not include any validation with other inputs.

Now less out of date and marginally easier to change in future. Testing is pretty lazy, most functionality is simply smoke tested on the various test configs. I am comfortable with this because i want to remove barriers to adding more in future such that this stays up to date.

Adds summary cli:
`mc summary tests/test_data/test_config_v14.xml`
image

image

Modifies debug cli:
`mc debug tests/test_data/test_config_v14.xml`
image

I think i've added most common problems but for sure there will be more found in future and i encourage people to either add or raise issues.

### CODE REVIEW

File: tests/test_debug.py [hunk 1]
```diff
@@ -11,61 +11,60 @@
...
+
+def test_get_modes():
+    assert BaseConfig(path=env.test_v14_xml_path).get_main_mode() == {"car"}
+    assert BaseConfig(path=env.test_v14_xml_path).get_mode_choices() == {'bus'  , 'train', 'tram', 'subway', 'walk', 'bike', 'train', 'ferry', 'car'}
+    assert BaseConfig(path=env.test_v14_xml_path).get_chained_modes() == {"car", "bike"}
+    assert BaseConfig(path=env.test_v14_xml_path).get_transit_modes() == {"bus", "tram", "train", "ferry", "subway", "helicopter"}
+    assert BaseConfig(path=env.test_v14_xml_path).get_deterministic_modes() == {"tram", "rail", "ferry", "subway"}
+    assert BaseConfig(path=env.test_v14_xml_path).get_intermodal_access_egress_modes() == {"car", "bike", "walk"}
+    assert BaseConfig(path=env.test_v14_xml_path).get_scoring_modes() == {'bus', 'pt', 'rail', 'tram', 'subway', 'walk', 'bike', 'ferry', 'car'}
+    assert BaseConfig(path=env.test_v14_xml_path).get_modes() == {'bus', 'pt', 'rail', 'train', 'tram', 'subway', 'walk', 'helicopter', 'bike', 'train', 'ferry', 'car'}
```
Comment (rorysedgwick) [inline tests/test_debug.py:25]: is there any cause for concern that `train` is listed twice in `get_modes()`?

File: mc/cli.py [hunk 2]
```diff
@@ -209,7 +209,7 @@ def build_config(
         acts=activities
     )
     if show:
-        config.print()
+        config.summary()
```
Comment (rorysedgwick) [inline mc/cli.py:212]: should `config.print()` on lines 139 & 161 also be updated to `config.summary()`?

**Review by rorysedgwick — APPROVED** [review]
This is beautiful.

I have been wondering about a way of automatically creating a run log of submitted simulations that contained just key info about the model parameters, and this is exactly the kind of summary I have been thinking of. Watch this space

**Review by akay-arup — APPROVED** [review]
👏🏼 👏🏼 👏🏼  This is rad. Thank you, Fred.

Do we want to add something about `maxEventQueueSize` in here as well? It's just the sort of annoying error that can ruin your day that this could help to avoid, but is also an edge case that's only arisen to date under super complex road pricing schemes.

**Pull request merged**